php-checks/src/test/resources/checks/StringLiteralDuplicatedCheck/default.php

```diff
@@ -53,3 +53,24 @@
 $score = $request->getParam('_score'); // OK - Only alphanumeric, -, _ and .
 $score = $request->getParam('_score');
 $score = $request->getParam('_score');
+
+$output = '';
+$output .= '<span class="payment">'.$total.'</span>';
+$output .= '<span class="payment-summary">'.$total_payment.'</span>';
+$output .= '<span class="badge normal">'.$inc_vat_text.'</span>';
+
+$output .= '<div>'.$total.'</div>';
+$output .= '<div>'.$total_payment.'</div>';
+$output .= '<div>'.$inc_vat_text.'</div>';
+
+$output = '<custom-tag_name:example.class>'.$var1.'</custom-tag_name:example.class>';
+$output = '<custom-tag_name:example.class>'.$var2.'</custom-tag_name:example.class>';
+$output = '<custom-tag_name:example.class>'.$var3.'</custom-tag_name:example.class>';
+
+$output .= '<div class="beautiful">'.$total.'</div>'; // Noncompliant {{Define a constant instead of duplicating this literal "<div class="beautiful">" 3 times.}}
+$output .= '<div class="beautiful">'.$total_payment.'</div>';
+$output .= '<div class="beautiful">'.$inc_vat_text.'</div>';
```

Comment:
I don't know if I would raise in this case. What makes it more worthwhile to create a constant for this? For any "HTML building" by concatenating, I would not raise 🤔 

Same for the case below, creating a constant for `</span` would not improve thing I think

Reply:
Good point about self-closing tags!
Ideally, we should not raise for cases like `<div class="beautiful">` or `</span`, the approach I implemented is about eliminating what we can 100% determine as HTML tag.
I extracted the first 10k issues on peach, I will work iteratively on this list to try to eliminate more complex use case, this rule seems very noisy on PHP (almost 60k issues)